js_modules/dagster-ui/packages/ui-core/src/assets/useWipeMaterializations.tsx

```diff
@@ -0,0 +1,52 @@
+import {MenuItem} from '@dagster-io/ui-components';
+import {useContext, useMemo, useState} from 'react';
+import {AssetWipeDialog} from 'shared/assets/AssetWipeDialog.oss';
+
+import {RefetchQueriesFunction} from '../apollo-client';
+import {CloudOSSContext} from '../app/CloudOSSContext';
+import {useUnscopedPermissions} from '../app/Permissions';
+
+export const useWipeMaterializations = ({
+  onComplete,
+  selected,
+  requery,
+}: {
+  onComplete?: () => void;
+  selected: {key: {path: string[]}}[];
+  requery?: RefetchQueriesFunction;
+}) => {
+  const [showWipeDialog, setShowWipeDialog] = useState(false);
+  const {
+    permissions: {canWipeAssets},
+  } = useUnscopedPermissions();
+
+  const {
+    featureContext: {canSeeWipeMaterializationAction},
+  } = useContext(CloudOSSContext);
+
+  const menuItem = useMemo(() => {
+    if (!canWipeAssets || !canSeeWipeMaterializationAction) {
+      return null;
+    }
```

Comment:
Maybe better to show the item but disable it in this case, perhaps with a tooltip? That way it's clear that the action is there, and why it can't be done in this circumstance.

Reply:
yeah I think I would prefer that too but i think we should do it for all the menu items (they all follow this pattern) at once.